clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -242,6 +377,60 @@ public AttributeList setAttributes(AttributeList list) {
             throw new UnsupportedOperationException("Set not allowed.");
         }
 
-    }
+        /**
+         * Add some new metrics to this mbean, then unregister and re-register the mbean.
+         */
+        void addMetrics(Collection<KafkaMetric> metrics) {
+            checkLocked();
+            if (!beanMetrics.isEmpty()) {
+                unregister();
+            }
+            for (KafkaMetric metric : metrics) {
+                beanMetrics.put(metric.metricName().name(), metric);
+            }
+            register();
```

Comment:
Are we assuming that `metrics` is non-empty?

Reply:
I was assuming that.  But on reflection, it's better to just check for that case and return early if so.